### PR TITLE
Add to playlist

### PR DESCRIPTION
**Pull request by Dananji**
Playlist dropdown with `Add new playlist` option:
image

Add to playlist with option to create playlist items for each structure item:
image

Add to playlist with option to add current track/time selection with custom title and description:
image

Display feedback from successful request:
image

Display message when the user doesn't have playlists:
image

### CODE REVIEW

File: app/views/media_objects/_add_to_playlist.html.erb [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+      }
+
+      function addListeners() {
+        $('#addToPlaylistPanel').on('show.bs.collapse', function (e) {
+          // Hide share resource panel on add to playlist panel collapse
+          $('#shareResourcePanel').collapse('hide');
+          // Hide add to playlist alert on panel collapse
+          $('#add_to_playlist_alert').slideUp(100);
+          // Show add to playlist form on collapse and reset initially
+          $('#add_to_playlist_form_group').slideDown();
```
Comment (cjcolvar) [inline app/views/media_objects/_add_to_playlist.html.erb:197]: I was confused at first by these comments.  What do you think about changing the wording?
```suggestion
          // Hide share resource panel on add to playlist panel show
          $('#shareResourcePanel').collapse('hide');
          // Hide add to playlist alert on panel show
          $('#add_to_playlist_alert').slideUp(100);
          // Show add to playlist form on show and reset initially
          $('#add_to_playlist_form_group').slideDown();
```

Comment (cjcolvar) [inline app/views/media_objects/_add_to_playlist.html.erb:197]: Should the slide reveal of the form happen at the end of this function after the reset has happened?

**Review by cjcolvar — COMMENTED** [review]
This looks great!  I like how you've added missing inline documentation comments for functions.  I had a couple real small questions but this looks pretty much ready to go.

File: app/views/media_objects/_add_to_playlist.html.erb (hunk 1 above)
Comment (Dananji) [reply app/views/media_objects/_add_to_playlist.html.erb]: Yes that sounds good. At this point I think the entire add to playlist panel is not showing, but it'd be safe to have this sliding down after everything is reset.

**Review by cjcolvar — APPROVED** [review]
Looks great!

**Pull request merged**